cht-petals/models.py

```diff
@@ -42,12 +42,12 @@ class PetalsBasedModel(ChatModel):
     def generate(
         cls,
         messages: list,
+        stop: Optional[Union[str, List[str]]],
```

Comment:
if this is optional then the default should be `None`, right?

```suggestion
        stop: Optional[Union[str, List[str]]] = None,
```

Also if using Python 3.10+:

```suggestion
        stop: str | list[str] | None = None,
```

Reply:
done :)

I didn't go with `|` syntax because currently we are using Optional in multiple places throughout the repo and `|` is not backward compatible too whereas `Optional` is